public/app/features/plugins/admin/pages/PluginDetails.test.tsx

```diff
@@ -53,25 +73,108 @@ function setup(pluginId: string): RenderResult {
 }
 
 describe('Plugin details page', () => {
-  it('should display install button for uninstalled plugins', async () => {
-    const { getByText } = setup('not-installed');
+  let dateNow: any;
 
-    const expected = 'Install';
+  beforeAll(() => {
+    dateNow = jest.spyOn(Date, 'now').mockImplementation(() => 1609470000000); // 2021-01-01 04:00:00
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  afterAll(() => {
+    dateNow.mockRestore();
+  });
+
+  it('should display an overview (plugin readme) by default', async () => {
+    const { queryByText } = setup('not-installed');
+    await waitFor(() => expect(queryByText(/licensed under the apache 2.0 license/i)).toBeInTheDocument());
+  });
 
-    await waitFor(() => expect(getByText(expected)).toBeInTheDocument());
+  it('should display version history', async () => {
+    const { queryByText, getByText, getByRole } = setup('installed');
+    await waitFor(() => expect(queryByText(/version history/i)).toBeInTheDocument());
+    userEvent.click(getByText(/version history/i));
+    expect(
+      getByRole('columnheader', {
+        name: /version/i,
+      })
+    ).toBeInTheDocument();
+    expect(
+      getByRole('columnheader', {
+        name: /last updated/i,
+      })
+    ).toBeInTheDocument();
+    expect(
+      getByRole('cell', {
+        name: /1\.0\.0/i,
+      })
+    ).toBeInTheDocument();
+    expect(
+      getByRole('cell', {
+        name: /5 years ago/i,
+      })
+    ).toBeInTheDocument();
   });
 
-  it('should not display install button for enterprise plugins', async () => {
+  it("should display install button for a plugin that isn't installed", async () => {
+    const { queryByRole } = setup('not-installed');
+
+    await waitFor(() => expect(queryByRole('button', { name: /install/i })).toBeInTheDocument());
+  });
+
+  it('should display uninstall button for an installed plugin', async () => {
+    const { queryByRole } = setup('installed');
+    await waitFor(() => expect(queryByRole('button', { name: /uninstall/i })).toBeInTheDocument());
+  });
+
+  it('should display update and uninstall buttons for a plugin with update', async () => {
+    const { queryByRole } = setup('has-update');
+
+    await waitFor(() => expect(queryByRole('button', { name: /update/i })).toBeInTheDocument());
+    expect(queryByRole('button', { name: /uninstall/i })).toBeInTheDocument();
+  });
+
+  it('should display install button for enterprise plugins if ', async () => {
+    config.licenseInfo.hasValidLicense = true;
     const { queryByRole } = setup('enterprise');
 
-    await waitFor(() => expect(queryByRole('button', { name: /(un)?install/i })).not.toBeInTheDocument());
+    await waitFor(() => expect(queryByRole('button', { name: /install/i })).not.toBeInTheDocument());
```

Comment:
Question: shouldn't we expect the `install` button to be visible in these cases?

```suggestion
    await waitFor(() => expect(queryByRole('button', { name: /install/i })).toBeInTheDocument());
```